sdk/core/azure-core/src/main/java/com/azure/core/spatial/CollectionGeometry.java

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.spatial;
+
+import java.util.Collection;
+import java.util.Collections;
+
+/**
+ * Represents a heterogeneous collection of {@link Geometry geometries}.
+ */
+public final class CollectionGeometry extends Geometry {
```

Comment:
Does the spec call it CollectionGeometry, or is this just your choice? GeometryCollection would make more sense otherwise?

Reply:
Also, is it weird to anyone else that a CollectionGeometry is a Geometry?

Reply:
It is a bit weird but it follows in the definition of the RFC. https://tools.ietf.org/html/rfc7946#section-3

Reply:
Ok - but you'll rename this to GeometryCollection?

Reply:
I think I'm going to keep the name as is for the reason that all other `Geometry` sub types are using the naming pattern `<Type>Geometry` (`PointGeometry`, `PolygonGeometry`). Switching this to `GeometryCollection` would cause it to be the only one which goes against that pattern.